src/graphql/data/util.test.tsx

```diff
@@ -0,0 +1,24 @@
+import { SupportedChainId } from 'constants/chains'
+
+import { Chain } from './__generated__/types-and-hooks'
+import { fromGraphQLChain, isSupportedGQLChain } from './util'
+
+describe('fromGraphQLChain', () => {
+  it('should return the corresponding chain ID for supported chains', () => {
+    expect(fromGraphQLChain(Chain.Ethereum)).toBe(SupportedChainId.MAINNET)
+
+    for (const chain of Object.values(Chain)) {
+      if (!isSupportedGQLChain(chain)) continue
+      expect(fromGraphQLChain(chain)).not.toBe(undefined)
+    }
+  })
+
+  it('should return undefined for unsupported chains', () => {
+    expect(fromGraphQLChain(Chain.UnknownChain)).toBe(undefined)
+
+    for (const chain of Object.values(Chain)) {
+      if (isSupportedGQLChain(chain)) continue
+      expect(fromGraphQLChain(chain)).toBe(undefined)
```

Comment:
https://github.com/Uniswap/interface/blob/main/cypress/README.md#avoid-branching-logic

These tests seem solid, but I think this recommendation applies here. It would be simpler to just provide two blocks, one with a chain that is supported, and a second with the unknown chain.

Reply:
the reason i added the loop was to allow the test to auto update as new supported chains are added

If we have a strong aversion to the branching logic I can make it more static with the tradeoff that when adding a new supported chain implementing engineer should add it to this test